data/pc/1.13/protocol.json

```diff
@@ -2016,6 +2016,18 @@
                       {
                         "name": "y",
                         "type": "i8"
+					  },
+                      {
+                        "name": "direction",
+                        "type": "i8"
+					  },
+                      {
+                        "name": "hasDisplayName",
+                        "type": "bool"
+                      },
+                      {
+                        "name": "displayName",
+                        "type": ["option", "string"]
```

Comment:
`hasDisplayName` is duplicated and should be removed. The `option` is implying there is a boolean flag to either having the next data type or not.

Reply:
`hasDisplayName` is a boolean value indicating whatever the map icon has a custom display name or not. See [here](https://wiki.vg/index.php?title=Pre-release_protocol#Map)

Reply:
You are right on the existence of this boolean, but `option` already imply there is such boolean. 
There are 2 equivalent ways to write such logic.
```
{
  "name": "displayName",
  "type": ["option", "string"]
}
```
or 
```
{
  "name": "hasDispalyName",
  "type": "bool"
},
{
  "name": "displayName",
  "type": [
    "switch",
    {
      "compareTo": "hasDisplayName",
      "fields": {"true": "string"},
      "default": "void"
    }
}
```

Reply:
I don't see why we should completely ignore a packet field just because it's redundant.

Reply:
nono, we did not ignore it.

Reply:
You said `hasDisplayName is duplicated and should be removed`. I could change the option to a switch but is that really necessary?

Reply:
Say we have a packet
```
01 02 03 04 05 
```
there are 5 bytes in total.
Assume boolean takes 1 byte and string takes 4 bytes.

```
{
  "name": "displayName",
  "type": ["option", "string"]
}
```
means the first byte is a boolean flag and the rest is a string. It correctly parses the packet and name the last 4 bytes to "displayName".

Whereas,
```
{
  "name": "hasDispalyName",
  "type": "bool"
},
{
  "name": "displayName",
  "type": ["option", "string"]
}
```
Means the first byte is a boolean flag and name it to `hasDispalyName`. The next byte is another boolean flag, and the rest of the bytes is a string but only 3 bytes left, so there is a mismatch.

Reply:
Ooooh I see, I didn't know the `option` type checked for a boolean, okay, I'll update the `protocol.json` without the `hasDisplayName`